assets/javascripts/job_next_previous.js

```diff
@@ -75,7 +75,7 @@ function renderJobResults(data, type, row) {
   // job status
   html += '<span id="res-' + row.id + '">';
   html += '<a href="' + urlWithBase('/tests/' + row.id) + '">';
-  if (row.result == 'none' && (row.state == 'running' || row.state == 'scheduled')) {
+  if (row.result == 'none' && (row.state === 'running' || row.state === 'scheduled' || row.state === 'new')) {
```

Comment:
can we keep the state "new" as a substate internally but treat it the same as scheduled for the whole UI layer including here?

Reply:
We could add yet another column to the jobs table (besides the relation for referencing Gru tasks) that allows us to flag jobs as not ready to be scheduled (without already having to assign a gru job ID).

Maybe we could also create a dummy Gru task without Minion job and assign that initially. This approach would avoid the extra column but also be a bit hacky.

Reply:
I was thinking about forward compatibility and options to extend for potential future use cases. job->state is already a flexible "varchar" in the schema so we can also reflect "new" there but here and in other cases we are in the UI layer so I thought that maybe we can translate from internal states to external states. We already have layed the groundwork with lib/OpenQA/Jobs/Constants.pm the past years where we have meta results and meta results. How about keeping "state" strictly internal – or maybe only reflect over the API for expert usage – and keep the UI layer strictly bound to the meta states?